pandas/core/internals.py

```diff
@@ -1052,6 +1052,11 @@ def where(self, other, cond, align=True, raise_on_error=True,
                 values = values.T
                 is_transposed = not is_transposed
 
+        if isinstance(other, string_types):
```

Comment:
maybe move this logic to ``core/common.py`` (and call the function here), something like ``_maybe_convert_string_to_object`` (possibly also see if code like this is duplicacted elsewhere)